src/RepoAuditor/Plugins/GitHubCustomization/Requirements/PullRequestTemplates.py

```diff
@@ -0,0 +1,82 @@
+# -------------------------------------------------------------------------------
+# |
+# |  Copyright (c) 2024 Scientific Software Engineering Center at Georgia Tech
+# |  Distributed under the MIT License.
+# |
+# -------------------------------------------------------------------------------
+"""Contains the PullRequestTemplate requirement."""
+
+from pathlib import Path
+from typing import Any
+
+from dbrownell_Common.Types import override
+
+from RepoAuditor.Requirement import EvaluateResult, ExecutionStyle, Requirement
+
+
+class PullRequestTemplate(Requirement):
+    """Validates that a pull request template is configured."""
+
+    def __init__(self) -> None:
+        super().__init__(
+            "PullRequestTemplate",
+            "Validates that a pull request template is configured.",
+            ExecutionStyle.Parallel,
+            "Create a pull request template file in one of these locations: .github/PULL_REQUEST_TEMPLATE.md, .github/pull_request_template.md, docs/PULL_REQUEST_TEMPLATE.md, or PULL_REQUEST_TEMPLATE.md",
+            "Pull request templates help standardize code contributions, ensuring consistent PR format and improving the review process.",
+        )
+
+    @override
+    def _EvaluateImpl(
+        self,
+        query_data: dict[str, Any],
+        requirement_args: dict[str, Any],  # noqa: ARG002
+    ) -> Requirement.EvaluateImplResult:
+        repo_path = query_data.get("repo_path")
+        if not repo_path:
+            return Requirement.EvaluateImplResult(
+                EvaluateResult.Error,
+                "Repository path not provided.",
+            )
```

Comment:
BTW, this is not consistent across the changes in this PR:
* Some classes have `repo_path = query_data["repo_path"]`, others have `repo_path = query_data.get("repo_path")`, with an extra check on `repo_path`.
* Some classes explicitly convert `repo_path` to a `PosixPath` object, others do not

Reply:
Agreed. This is all addressed in Part 3. 🙂